### PR TITLE
Feature/asep 13

### PR DESCRIPTION
**Pull request by saurabhprasun20**
PR for ASEP-13. Need to add two more tests later. I will create PR for that along with swagger task.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -2,6 +2,10 @@ version: '3.8'
 services:
     mongodb_airbnb:
         image: mongo
+        container_name: airbnb-database
+        environment:
+        - PUID=1000
+        - PGID=1000
```
Comment (niels89) [inline docker-compose.yml:8]: This is not needed anymore, I included it in the first version, but it works fine without it.

File: eloquent-elks-backend/src/main/resources/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    container_name: "airbnb-database"
+    ports:
+      - 27017:27017
+  springboot-mongodb:
+    image: springboot-mongodb:1.0
+    container_name: springboot-mongodb
+    ports:
+      - 8080:8080
+    links:
+      - airbnb-database
```
Comment (niels89) [inline eloquent-elks-backend/src/main/resources/docker-compose.yml:14]: What's the purpose of this second docker compose file?

File: eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/DTO/PropertyFecthDTOTest.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
+package com.airbnb.eloquentelksbackend.DTO;
+
+import com.airbnb.eloquentelksbackend.entity.Property;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PropertyFecthDTOTest {
```
Comment (niels89) [inline eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/DTO/PropertyFecthDTOTest.java:8]: Typo in class name: Property**Fecth**DTOTest

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/DTO/DTOMapper.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    DTOMapper INSTANCE = Mappers.getMapper(DTOMapper.class);
+
+    @Mapping(source = "name", target="name")
+    @Mapping(source = "longitude", target="longitude")
+    @Mapping(source = "latitude", target="latitude")
+    @Mapping(source = "roomType", target="roomType")
+    PropertyFetchDTO convertPropertyToPropertyFetchDTO(Property property);
+
+    List<PropertyFetchDTO> map(List<Property> employees);
+}
```
Comment (niels89) [inline eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/DTO/DTOMapper.java:22]: Please add javadoc comments to describe your classes and instance variables

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/DTO/PropertyFetchDTO.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+@Getter
+@Setter
+@ToString
+public class PropertyFetchDTO {
+    private String name;
+    private double longitude;
+    private double latitude;
+    private String roomType;
+}
```
Comment (niels89) [inline eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/DTO/PropertyFetchDTO.java:15]: Please add javadoc comments to describe your classes and instance variables

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/controller/PropertyController.java [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+
+    @GetMapping("/airbnb")
+    @ResponseStatus(HttpStatus.OK)
+    @ResponseBody
+    public List<PropertyFetchDTO> getProperty(){
+        List<Property> propertiesList = propertyService.getAllProperties();
+        return DTOMapper.INSTANCE.map(propertiesList);
+    }
```
Comment (niels89) [inline eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/controller/PropertyController.java:34]: Please add javadoc comments to describe your classes and instance variables

File: eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplicationTests.java [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class EloquentElksBackendApplicationTests {
+
+	@Test
+	void contextLoads() {
+	}
+
+}
```
Comment (niels89) [inline eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplicationTests.java:13]: This test doesn't really do anything, do we need this?

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/service/PropertyService.java [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+        this.propertyRepository = propertyRepository;
+    }
+
+    public List<Property> getAllProperties(){
+        List<Property> allEntries = propertyRepository.findAll();
+        return allEntries;
+    }
+
+
+}
```
Comment (niels89) [inline eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/service/PropertyService.java:29]: Please add javadoc comments to describe your classes and instance variables

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/repository/PropertyRepository.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
+
+
+import java.util.List;
+
+@Repository("userRepository")
+public interface PropertyRepository extends MongoRepository<Property, Long> {
+    List<Property> findAll();
+}
```
Comment (niels89) [inline eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/repository/PropertyRepository.java:13]: Please add javadoc comments to describe your classes and instance variables

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplication.java [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+	}
+
+	@GetMapping(value = "/", produces = MediaType.TEXT_PLAIN_VALUE)
+	@ResponseStatus(HttpStatus.OK)
+	@ResponseBody
+	public String helloWorld() {
+		return "The application is running.";
+	}
+
+}
```
Comment (niels89) [inline eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplication.java:28]: Please add javadoc comments to describe your classes and instance variables

**Review by niels89 — CHANGES_REQUESTED** [review]

**Comment by niels89** [comment]
Hey @saurabhprasun20, thanks for your work. Looks good in general, I have made some comments on a few details that I think should be changed. A group of comments are about javadoc documentation. I didn't mark all the instances in the code where javadoc is missing, so you might need to check if you find other classes that are undocumented.

File: docker-compose.yml (hunk 1 above)
Comment (saurabhprasun20) [reply docker-compose.yml:8]: Then I will remove it and then merge

File: eloquent-elks-backend/src/main/resources/docker-compose.yml (hunk 2 above)
Comment (saurabhprasun20) [reply eloquent-elks-backend/src/main/resources/docker-compose.yml:14]: This is for my local testing only

File: eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/DTO/PropertyFecthDTOTest.java (hunk 3 above)
Comment (saurabhprasun20) [reply eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/DTO/PropertyFecthDTOTest.java:8]: Changed

File: eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplicationTests.java (hunk 7 above)
Comment (saurabhprasun20) [reply eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplicationTests.java:13]: It was default.

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/DTO/DTOMapper.java (hunk 4 above)
Comment (niels89) [reply eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/DTO/DTOMapper.java:22]: Will be done in ASEP-27

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/DTO/PropertyFetchDTO.java (hunk 5 above)
Comment (niels89) [reply eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/DTO/PropertyFetchDTO.java:15]: Will be done in ASEP-27

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/controller/PropertyController.java (hunk 6 above)
Comment (niels89) [reply eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/controller/PropertyController.java:34]: Will be done in ASEP-27

File: eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplicationTests.java (hunk 7 above)
Comment (niels89) [reply eloquent-elks-backend/src/test/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplicationTests.java]: Will be done in ASEP-27

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/service/PropertyService.java (hunk 8 above)
Comment (niels89) [reply eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/service/PropertyService.java:29]: Will be done in ASEP-27

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/repository/PropertyRepository.java (hunk 9 above)
Comment (niels89) [reply eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/repository/PropertyRepository.java:13]: Will be done in ASEP-27

File: eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplication.java (hunk 10 above)
Comment (niels89) [reply eloquent-elks-backend/src/main/java/com/airbnb/eloquentelksbackend/EloquentElksBackendApplication.java:28]: Will be done in ASEP-27

**Review by niels89 — APPROVED** [review]
Looks good now, Javadoc will be added in ASEP-27

**Pull request merged**